packages/neuron-wallet/src/block-sync-renderer/sync/light-connector.ts

```diff
@@ -232,14 +233,32 @@ export default class LightConnector extends Connector<CKBComponents.Hash> {
       })
       return
     }
-    this.transactionsSubject.next({ txHashes: result.txs.map(v => v.txHash), params: syncProgress.hash })
+    const txHashes = result.txs.map(v => v.txHash)
+    await this.fetchPreviousOutputs(txHashes)
+    this.transactionsSubject.next({ txHashes, params: syncProgress.hash })
     this.syncInQueue.set(syncProgress.hash, {
       blockStartNumber: result.lastCursor === '0x' ? parseInt(blockRange[1]) : parseInt(blockRange[0]),
       blockEndNumber: parseInt(blockRange[1]),
       cursor: result.lastCursor === '0x' ? undefined : result.lastCursor,
     })
   }
 
+  private async fetchPreviousOutputs(txHashes: string[]) {
+    const transactions = await this.lightRpc
+      .createBatchRequest<'getTransaction', string[], TransactionWithStatus[]>(txHashes.map(v => ['getTransaction', v]))
+      .exec()
+    const previousTxHashes = new Set<string>()
+    transactions.forEach(tx => {
+      tx.transaction.inputs.forEach(input => {
+        const previousTxHash = input.previousOutput!.txHash
+        if (previousTxHash !== `0x${'0'.repeat(64)}`) {
+          previousTxHashes.add(previousTxHash)
+        }
+      })
+    })
```

Comment:
`transactions.flatMap(tx => tx.inputs).forEach(input => {})` is easy to understand.